arch/arm/core/aarch64/switch.S

```diff
@@ -34,13 +33,12 @@ GDATA(_k_neg_eagain)
 GTEXT(z_arm64_context_switch)
 SECTION_FUNC(TEXT, z_arm64_context_switch)
 #ifdef CONFIG_TRACING
+	stp	x0, x1, [sp, #-16]!
```

Comment:
I did not understand the reason for save restore of x0, x1 here

Reply:
`x0` and `x1` can be clobbered by `sys_trace_thread_switched_out`. At the same time I cannot load `x0` and `x1` from the stack after tracing because `z_arm64_context_switch()` is also called by https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/arm/core/aarch64/isr_wrapper.S#L98 where `x0` and `x1` on the stack are unrelated values.

Reply:
That is prefect.  I missed to expand and see the usage of these registers.